plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -703,6 +703,10 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
 
+        if (table.getSnapshotId().isPresent() && table.getSnapshotId().get() != icebergTable.currentSnapshot().snapshotId()) {
+            throw new TrinoException(NOT_SUPPORTED, "Inserting data into snapshot is not supported in Iceberg.");
```

Comment:
The current snapshot is supported, so how about changing to "Inserting data into past snapshot..." or something?